reframe/frontend/statistics.py

```diff
@@ -216,8 +217,9 @@ def print_failure_report(self, printer):
             printer.info(f"  * System partition: {r['system']}")
             printer.info(f"  * Environment: {r['environment']}")
             printer.info(f"  * Stage directory: {r['stagedir']}")
-            nodelist = ','.join(r['nodelist']) if r['nodelist'] else None
-            printer.info(f"  * Node list: {nodelist}")
+            printer.info(
+                f"  * Node list: {util.nodelist_abbrev(r['nodelist'])}"
```

Comment:
Is it guaranteed that the `r['nodelist']` will always contain an entry? Because there are assertion points down the road.
Previously there a check whether the `r['nodelist']` was not empty.

Reply:
Good point, I'll have a look, but I think that it has to be a list according to the JSON schedma.

Reply:
It is guaranteed to be a list. Check here:

https://github.com/eth-cscs/reframe/blob/e8a81b3191b2443973562c621cd432611656659a/reframe/frontend/statistics.py#L114

And here

https://github.com/eth-cscs/reframe/blob/e8a81b3191b2443973562c621cd432611656659a/reframe/frontend/statistics.py#L142

And nodelist must a be a list in the JSON report:

https://github.com/eth-cscs/reframe/blob/e8a81b3191b2443973562c621cd432611656659a/reframe/schemas/runreport.json#L50-L52